src/lib/y2partitioner/widgets/overview.rb

```diff
@@ -0,0 +1,246 @@
+require "cwm/widget"
+
+require "y2partitioner/icons"
+
+Yast.import "Hostname"
+
+module Y2Partitioner
+  module Widgets
+    # Widget representing partitioner overview.
+    #
+    # It has replace point where it displays more details about selected element
+    # in partitioning.
+    #
+    # TODO: abstract treewidget from it
+    class Overview < CWM::CustomWidget
+      # creates new widget for given device graph
+      def initialize(device_graph)
+        self.handle_all_events = true
+        @opened = [:all]
+        @device_graph = device_graph
+      end
+
+      # content of widget
+      def contents
+        Tree(Id(:tree), Opt(:notify), _("System View"), items)
+      end
+
+      # handles widgets. As it is with notify, it will get any click on Item
+      def handle(event)
+        id = event["ID"]
+        log.info "handling id #{id}"
+
+        nil
+      end
+
+    private
+
+      def items
+        [
+          Item(
+            Id(:all),
+            term(:icon, Icons::ALL),
+            open?(:all),
+            "machine", # TODO: stuck getting hostname on my pc Yast::Hostname.CurrentHostname,
+            machine_items
+          ),
+          # TODO: only if there is graph support UI.HasSpecialWidget(:Graph)
+          device_graph_item,
+          mount_graph_item,
+          summary_item,
+          settings_item
+        ]
+      end
+
+      def device_graph_item
```

Comment:
Maybe you could have some `#widget_item` (or maybe a better name :) ) which receives an Id and an icon to reduce duplication:

```ruby
widget_item(:device_graph, _("Device Graph"), Icons::GRAPH)
widget_item(:mountgraph, _("Mount Graph"), Icons::GRAPH)
widget_item(:summary, _("Installation Summary"), Icons::SUMMARY)
```

On the other hand it is still a WIP, so I guess you (or anyone) can do it later.